numpy/core/src/multiarray/shape.c

```diff
@@ -277,9 +278,6 @@ PyArray_Newshape(PyArrayObject *self, PyArray_Dims *newdims,
         goto fail;
     }
 
-    if (incref) {
-        Py_INCREF(self);
-    }
     if (PyArray_SetBaseObject(ret, (PyObject *)self)) {
         Py_DECREF(ret);
         return NULL;
```

Comment:
missing decref, better as `goto fail`

Reply:
`PyArray_SetBaseObject` steals a reference, so this is correct as is